zerver/views/message_fetch.py

```diff
@@ -873,7 +914,8 @@ def get_messages_backend(request: HttpRequest, user_profile: UserProfile,
     if anchored_to_right:
         num_after = 0
 
-    first_visible_message_id = get_first_visible_message_id(user_profile.realm)
+    first_visible_message_id = get_first_visible_message_id(user_realm)
```

Comment:
`find_first_unread_anchor` is probably not going to work with an anonymous user; can you add a test for passing `anchor="first_unread"` with this feature?  I think the outcome should be that for these web-public views, it always does the same thing as `anchored_to_right`.